### ISSUE TITLE
Feature Request: Cascade Delete

### ISSUE THREAD

**Issue by CaffeinatedFunctionality** [arangodb/arangodb#12530]
## My Environment

* __ArangoDB Version__:        3.7.1
* __Storage Engine__:             RocksDB
* __Deployment Mode__:        Single Serve
* __Deployment Strategy__:    Manual Start in Docker
* __Infrastructure__:               own
* __Operating System__:        Windows 10
* __Total RAM in your machine__:        32Gb
* __Disks in use__:                 SSD
* __Used Package__:              Docker - official Docker library

I'm working on an app and I have sort of a tiered environment like the following diagram

(Document1) ===> [edge1] ===> (Document2) ===> [edge2] ===> (Document3)

What I need to do is Remove or Delete Document 2, but I would also need to delete edge 1 along with everything down the chain related to it. I'm searching in the documentation and it doesn't seem like something I can easily figure out either that or I'm going to have what I consider a very complex query just to delete. I haven't built named graphs yet, and it seems there is an ok way to delete the edges if I build a named graph and handle it that way, but that seems to only delete the edges. I could do this same thing for the nodes and delete the Documents in a loop, but that seems like a ton of work to do a cascade delete. Is there any way to look into some kind of functionality that deletes all dependent edges and documents? or maybe something we could make delete this document and also these named documents and edges related to it? 

Hopefully that sounds clear. It seems uber complicated at the moment to delete a document with a lot of dependent relationships.

**Comment by CaffeinatedFunctionality** [comment]
I probably should also mention I am using the nodeJS driver

**Comment by dothebart** [comment]
You're right. at the moment there is just this API to delete it in a subsequent call:
 https://www.arangodb.com/docs/stable/graphs-general-graphs-management.html#remove-a-vertex

In AQL you would need manual plumbing.

**Comment by Simran-B** [comment]
As Willi said, the Graph API can take care of that. In AQL this isn't implemented and there were concerns about performance that spoke against supporting it - but it's unclear whether this is still true with RocksDB.

Anyway, maybe this is helpful:
https://www.arangodb.com/docs/stable/aql/examples-remove-vertex.html

**Comment by CaffeinatedFunctionality** [comment]
The problem I am seeing, though it could be because I am a newb at Arango, is named graphs can only have one defined edge, in my example, edge1 would be a different type of edge than edge2 and couldn't be in the same named graph. I also need to delete the child document and not just the edges which is the problem. I worked with Neo4J before and it had a detach delete which removes the edges and nodes. It worked quite well and I'm looking for that style of functionality. Though now that I think about it I wonder if there is a way to use the remove vertices and create a garbage manager to remove orphan documents with missing dependent parents.

**Comment by CaffeinatedFunctionality** [comment]
So to delete the documents as well as the paths, which is why I made this post, would it make sense to do something like

LET edgeKeys = (FOR v,e,p IN 1..1 ANY 'documents/document1' GRAPH 'doc_to_doc'
REMOVE e._key in edges
FOR vert IN p.vertices 
REMOVE v._key in documents)
